### PR TITLE
PRC-134 Replace Govuk Pagination With MOJ Version

### PR DESCRIPTION
**Pull request by rosariodiaferia-moj**
**Ticket**: [PRC-134](https://dsdmoj.atlassian.net/browse/PRC-134)

**Description**: Govuk pagination replaced with Moj version

Hmpps-Contacts-Ui-List-contacts-10-21-2024_08_03_PM

[PRC-134]: https://dsdmoj.atlassian.net/browse/PRC-134?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: server/routes/contacts/manage/contact-search/contactSearchController.test.ts [hunk 1]
```diff
@@ -256,40 +256,135 @@ describe('Contact seaarch results', () => {
...
-      totalElements: 15,
-      totalPages: 2,
-    }
-    contactsService.searchContact.mockResolvedValue(results)
+  describe('Pagination', () => {
+    it('should display the pagination when total pages are more than 1', async () => {
+      // Given
+      existingJourney = {
+        ...existingJourney,
+        searchContact: {
```
Comment (prabash-moj) [inline server/routes/contacts/manage/contact-search/contactSearchController.test.ts:264]: would it be possible to move this section into a private data set up method and reuse in 3 tests to remove code duplication?
````
   existingJourney = {
        ...existingJourney,
        searchContact: {
          contact: { lastName: 'last', middleNames: '', firstName: '' },
          dateOfBirth: { day: undefined, month: undefined, year: undefined },
        },
      }
      auditService.logPageView.mockResolvedValue(null)
```
      prisonerSearchService.getByPrisonerNumber.mockResolvedValue(TestData.prisoner())

      const contactsArray = []
      for (let i = 0; i < 15; i += 1) {
        contactsArray.push(TestData.contactSearchResultItem({ id: i }))
      }`

File: server/routes/contacts/manage/prisoner-search/prisonerSearchResultsController.test.ts [hunk 2]
```diff
@@ -47,6 +48,92 @@ afterEach(() => {
 })
 
 describe('GET /contacts/manage/prisoner-search-results', () => {
+  describe('Pagination', () => {
+    it('should display the pagionation when total pages are more than 1', async () => {
```
Comment (prabash-moj) [inline server/routes/contacts/manage/prisoner-search/prisonerSearchResultsController.test.ts:52]: typo on pagionation ->pagination

**Review by prabash-moj — APPROVED** [review]

File: server/routes/contacts/manage/contact-search/contactSearchController.test.ts (hunk 1 above)
Comment (rosariodiaferia-moj) [reply server/routes/contacts/manage/contact-search/contactSearchController.test.ts:264]: Changes added

File: server/routes/contacts/manage/prisoner-search/prisonerSearchResultsController.test.ts (hunk 2 above)
Comment (rosariodiaferia-moj) [reply server/routes/contacts/manage/prisoner-search/prisonerSearchResultsController.test.ts]: typo corrected

**Review by prabash-moj — APPROVED** [review]

**Pull request merged**